cluster/src/test/scala/org/apache/spark/sql/store/BugTest.scala

```diff
@@ -1202,4 +1202,20 @@ class BugTest extends SnappyFunSuite with BeforeAndAfterAll {
     assert(metrics(numRowsMetric.accumulatorId) ===
         SQLMetrics.stringValue(numRowsMetric.metricType, numRows :: Nil))
   }
+
+  test("SNAP-3141: code gen failure") {
+    // TODO: new SHA code generation fails for query below
+    val session = snc.snappySession.newSession()
+    session.sql(s"set ${Property.UseOptimzedHashAggregate.name} = true")
+    session.sql(s"set ${Property.UseOptimizedHashAggregateForSingleKey.name} = true")
+
+    val numRows = 1000000
+
+    session.sql("create table test1 (id long, data string) using column " +
+      s"options (buckets '8') as select id, 'data_' || id from range($numRows)")
+    val ds = session.sql(
+      "select avg(id) average, id % 10 from test1 group by id % 10 order by average")
+
+    ds.collect()
+  }
```

Comment:
Instead of adding new test you can just remove/change the set calls in the previous test for SNAP-3123 and add a comment there that it also tests for SNAP-3141 (or name of test can be like "SNAP-3123_3141: check GUI plans and codegen failure")